archive/frames/models.py

```diff
@@ -159,6 +161,69 @@ def as_dict(self):
         ret_dict['related_frames'] = list(self.related_frames.all().values_list('id', flat=True))
         return ret_dict
 
+
+class Thumbnail(models.Model):
+    frame = models.ForeignKey(
+        Frame, 
+        on_delete=models.CASCADE, 
+        related_name='thumbnails', 
+        null=True,
+        blank=True,
+        help_text="The frame this thumbnail is associated with"
+    )
+    size = models.CharField(
+        max_length=20,
+        help_text="String description of the size of the thumbnail"
+    )
+    basename = models.CharField(
+        max_length=1000, 
+        db_index=True, 
+        unique=True,
+        help_text="The basename of the thumbnail"
+    )
+    extension = models.CharField(
+        max_length=20,
+        help_text="The file extension of the thumbnail"
+    )
+    key = models.CharField(
+        max_length=32,
+        help_text="The key used to store the thumbnail in the file store",
+        default=''
+    )
+
+    def as_dict(self):
+        ret_dict = model_to_dict(self)
+        ret_dict['url'] = self.url
+        ret_dict['size'] = self.size
+        return ret_dict
+    
+    @property
+    def filename(self):
+        """
+        Returns the full filename for the thumbnail
+        """
+        return '{0}{1}'.format(self.basename, self.extension)
+     
+    @cached_property
+    def url(self):
+        metadata = self.frame.get_header_dict()
+        # include frame basename and size so that this passes metadata validation in the DataFile class
```

Comment:
I think the proper approach would be to add a thumbnail type to the ocs_archive library - we might want to store thumbnails in their own /thumbnails directory in the filestore, not in raw or processed like it would now. This could also be done in the ocs_archive file type we create for thumbnails. It should also not need to use the related frames basename then, just its own basename and the frame fields (to get site/inst/dayobs/ directory) is probably enough. Also shouldn't need to get the header, which would make another DB call or table join - it just needs the site, inst, dayobs to determine its directory if we set that up in the ocs_archive thumbnail file class, and it can get those directly from the frame model.